### ISSUE TITLE
Proposal to prioritize The Merge after London

### ISSUE THREAD

**Issue by mkalinin** [ethereum/pm#278]
With the recent Vitalik's quick merge proposal there is a possibility to deliver the merge sooner. The main advantage of the proposal is that it introduces fewer required changes to make the consensus upgrade on Ethereum Mainnet, i.e. replace PoW with PoS.

This proposal does imply that the Merge becomes the main focus of Ethereum core developers after London and Altair (the beacon chain upgrade expected to happen soon).

### Rationale

The Merge is the main thing that the Ethereum community expects, and working on other initiatives (i.e. another “feature fork”) takes the limited focus that core developers have away from the merge.

### Quick merge proposal
- https://notes.ethereum.org/m9IX3OkkTveXCCOSzGkUiw

**Comment by eggswap** [comment]
Funny it took miners working together to inspire such haste.

**Comment by chmichael** [comment]
> Funny it took miners working together to inspire such haste.

I'm very curious to see how fast or slow the network will be with PoS only. (iow without sharding)

**Comment by jonscherman** [comment]
I Think the priority should be ETH 2. A lot of new blockchain like Polkadot or Cordano are comming and if eth is too long to implement eth 2, these other blockchain will surpass eth

**Comment by ibelite2** [comment]
Can someone ELI5 the benefit to the user after the consensus switch from POW to POS?

1. Will all transactions be done by ETH 2.0 validators and therefore reduce cost of transacting on ETH?
2. How will this affect smart contract computation (e.g. DeFi apps)? Will all EVM interactions happen on POS?

**Comment by mkalinin** [comment]
> @timbeiko @mkalinin
> 
> Can someone ELI5 the benefit to the user after the consensus switch from POW to POS?
> 
> 1. Will all transactions be done by ETH 2.0 validators and therefore reduce cost of transacting on ETH?
> 2. How will this affect smart contract computation (e.g. DeFi apps)? Will all EVM interactions happen on POS?

1. Consensus upgrade has no direct impact on transaction fees, transaction fees for the end user are expected to become lower when rollups start working under PoS and using shards. By swapping PoW with PoS the application layer gains native access to the data availability layer provided by shards. Then rollups being an application layer solutions leverage consensus upgrade + shards to deliver a significantly higher throughput and lower fees
2. All EVM interactions will be secured by PoS and in general it does not affect smart contract computations. Other than `BLOCKHASH` seem to loose its strength as a source of randomness and apps will get access to a stronger entropy source, i.e. randao. Composability and contract interactions, JSON-RPC remain the same as on the mainnet today. The other property of PoS that might matter here is more predictable block times (the beginning of each slot), i.e. there is no fluctuations related to Poisson process like on the PoW chain

**Comment by timbeiko** [comment]
Linking @mkalinin's latest doc on this for reference: https://hackmd.io/@n0ble/ethereum_consensus_upgrade_mainnet_perspective

**Comment by timbeiko** [comment]
Closing this because we agreed on ACD 111 to focus on the Merge in parallel to London and prioritize shipping it as soon as it is ready.
